cmake/modules/AddSwift.cmake

```diff
@@ -854,6 +848,31 @@ function(_add_swift_library_single target name)
     set(SWIFTLIB_SINGLE_API_NOTES "${module_name}")
   endif()
 
+  # On Linux and FreeBSD add conformance magic objects that coalesce protocol
+  # conformance sections. This wouldn't be necessary if the link was done by
+  # the swift binary: rdar://problem/19007002
+  if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" OR
+     "${CMAKE_SYSTEM_NAME}" STREQUAL "FreeBSD")
+
+    if("${libkind}" STREQUAL "SHARED")
+      set(arch_subdir "${SWIFTLIB_DIR}/${SWIFTLIB_SINGLE_SUBDIR}")
+
+      add_custom_command(
```

Comment:
Please use `add_custom_command_target`.  `add_custom_command` is not safe in certain (subtle) scenarios in parallel builds.

```
    "Do not list the output in more than one independent target that
     may build in parallel or the two instances of the rule may
     conflict (instead use add_custom_target to drive the command and
     make the other targets depend on that one)."
```